### PR TITLE
docs: cluster observability documentation and dashboard improvements

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-293] 

## Description

Merging Cluster Observability feature branch to main. New instructions for setting up observability in Kubernetes environments, along with updated dashboards.

## Test Plan

None needed.

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-293]: https://hpe-aiatscale.atlassian.net/browse/RM-293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/release-notes/observability.st [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:orphan:
+
+**Improvements**
+
+Kubernetes: Add Determined resource information such as `workspace` and `task ID` as pod labels. This improvement facilitates better resource tracking and management within Kubernetes environments.
+
+Configuration: Introduce a DCGM Helm chart and Prometheus configuration to the `tools/observability` directory. Additionally, two new dashboards, "API Monitoring" and "Resource Utilization", have been added to improve observability and operational insight.
```
Comment (tara-det-ai) [inline docs/release-notes/observability.st:7]: this is an "st" file, please change it to an *.rst file

"docs/release-notes/observability.st" should be "docs/release-notes/observability.rst"

Comment (kkunapuli) [reply docs/release-notes/observability.st:7]: Good catch - thank you!

File: docs/integrations/prometheus/_index.rst [hunk 2]
```diff
@@ -16,11 +16,14 @@
...
 
-Discover how to enable a Grafana dashboard to monitor Determined hardware and system metrics on a
-cloud cluster, such as AWS or Kubernetes. Determined provides a Prometheus endpoint that contains
-mappings between internal task, GPU, and container definitions, which are used by Prometheus to
-collect relevant metrics on a cluster running Determined. The endpoint is not enabled by default but
-can be enabled in the master configuration file.
+Discover how to enable a Grafana dashboard to monitor Determined hardware and system metrics.
+Determined provides a Prometheus endpoint that contains mappings between internal task, GPU, and
+container definitions, which are used by Prometheus to collect relevant metrics on a cluster running
+Determined. The endpoint is not enabled by default but can be enabled in the master configuration
```
Comment (NicholasBlaskey) [inline docs/integrations/prometheus/_index.rst:22]: prometheus should be enabled by default now

**Review by NicholasBlaskey — APPROVED** [review]
LGTM

File: docs/integrations/prometheus/_index.rst (hunk 2 above)
Comment (kkunapuli) [reply docs/integrations/prometheus/_index.rst:22]: Nice catch! Thank you!

I inverted the sentence `The endpoint is enabled by default but can be disabled in the master configuration
file.`

**Pull request merged**